Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -285,25 +272,27 @@ public void GarbageCollect(OperationContext context)
 
             using (var cancellableContext = TrackShutdown(context))
             {
-                DoGarbageCollect(cancellableContext);
+                var tasks = new List<Task> { Task.Run(() => DoGarbageCollectMetadata(cancellableContext)) };
+                if (_isContentGarbageCollectionEnabled)
+                {
+                    tasks.Add(Task.Run(() => DoGarbageCollectContent(cancellableContext)));
+                }
+                Task.WaitAll(tasks.ToArray());
             }
 
             lock (this)
             {
                 if (!ShutdownStarted)
                 {
-                    if (_isGarbageCollectionEnabled)

```

Comment:
It seems that after this change we'll keep setting the timer even when the GC is disabled. Right?

Reply:
GC is never disabled after this change. Only Content GC may be disabled, and that doesn't affect Metadata GC

---
In reply to: [309383052](https://github.com/microsoft/BuildXL/pull/679#discussion_r309383052) [](ancestors = 309383052)